framework/web/DbSession.php

```diff
@@ -165,12 +165,15 @@ public function writeSession($id, $data)
         // exception must be caught in session write handler
         // http://us.php.net/manual/en/function.session-set-save-handler.php#refsect1-function.session-set-save-handler-notes
         try {
-            $query = new Query();
-            $exists = $query->select(['id'])
-                ->from($this->sessionTable)
-                ->where(['id' => $id])
-                ->createCommand($this->db)
-                ->queryScalar();
+            // Fix for #11401, concurrency when inserting new session ID            
+            $exists = $this->db->useMaster(function($db) use ($id) {
+                return (new Query())->select(['id'])
+                    ->from($this->sessionTable)
+                    ->where(['id' => $id])
+                    ->createCommand($db)
+                    ->queryScalar();
```

Comment:
Shouldn't we replace these two lines with `->exists($id);`?

Reply:
it should be `->exists($this->db)`